### ISSUE TITLE
Support aligning AS keyword in column aliases 

### ISSUE THREAD

**Issue by amardeep** [sqlfluff/sqlfluff#1582]
Tools such as Datagrip allow aligning AS keyword in column aliases. For eg., given input sql:
```sql
SELECT
  customer_id AS id,
  customer_name AS name,
  age
FROM customers
WHERE age < 60
```

it would reformat it to:
```sql
SELECT
  customer_id   AS id,
  customer_name AS name,
  age
FROM customers
WHERE age < 60
```

In most of our queries, we find this makes the file more readable.
It would be great if sqlfluff could support it to.

**Comment by jpy-git** [comment]
Feels like this should be an option of L003, either align in multiples of indents from the left hand side of the page (current) or align on as (this proposal)

**Comment by christopherridley** [comment]
Doesn't seem like L003 makes sense to me here. L003 is great to make sure the column names are aligned appropriately. L039 to ignore additional whitespace directly before the "AS" only between "SELECT" and "FROM" would be ideal here I believe. That would give people the flexibility to align the column aliases or not if they wish.

**Comment by Olaktal** [comment]
+1 we align all our aliases and I think it makes SQL clearer 👌 
Any update on this feature ? Maybe it could be a configuration settings 🤔

**Comment by RoyalTS** [comment]
Would love this (as an option to L039?) as well. Makes for much cleaner code!

**Comment by christopherridley** [comment]
Seems like it doesn't really need to even be an option. I think if the logic is added to L039 to ignore additional whitespace directly before the "AS" only between "SELECT" and "FROM", it allows for people to do both styles. 

select
        company_id                      as company_id,
        deal_id                              as deal_id,
        _fivetran_synced             as _fivetran_synced

from source

or

select
        company_id as company_id,
        deal_id as deal_id,
        _fivetran_synced as _fivetran_synced

from source

**Comment by moseleyi** [comment]
This is one of my main rules to do this when possible to increase the readability of the file!
